pkg/controller/vitesskeyspace/reconcile_topo.go

```diff
@@ -56,38 +55,21 @@ func (r *ReconcileVitessKeyspace) reconcileTopology(ctx context.Context, vtk *pl
 	defer ts.Close()
 
 	if *vtk.Spec.TopologyReconciliation.PruneShards {
-		result, err := r.pruneShards(ctx, vtk, ts)
-		resultBuilder.Merge(result, err)
-	}
-
-	return resultBuilder.Result()
-}
-
-func (r *ReconcileVitessKeyspace) pruneShards(ctx context.Context, vtk *planetscalev2.VitessKeyspace, ts *toposerver.Conn) (reconcile.Result, error) {
-	resultBuilder := &results.Builder{}
+		desiredShards := make(sets.String, len(vtk.Status.Shards))
```

Comment:
If you agree with the below comment, then this `desiredShards` logic could be entirely isolated inside the prune_shards.go file.

Reply:
Desired shards might be calculated in different ways by consumers of `ShardsToPrune`, and this PR hopes to make `ShardsToPrune` and `DeleteShards` as easy to use by external consumers as possible.